imap/mailbox.c

```diff
@@ -7915,6 +7953,16 @@ EXPORTED struct mailbox_iter *mailbox_iter_init(struct mailbox *mailbox,
     iter->num_records = mailbox->i.num_records;
     iter->msg = message_new();
 
+    /* Are we iterating forward or backward? */
+    if (flags & ITER_STEP_BACKWARD) {
```

Comment:
ooh nice - this is a handy API.  We will need to make sure it's something we can do with any new tech we port this to though